File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/mf_orders.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json     = Json(JsonConfiguration.Stable)
// val mFOrders = json.parse(MFOrders.serializer(), jsonString)

package MfOrders

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class MFOrders (
    val data: List<Datum>? = null,
    val status: String? = null
)

@Serializable
data class Datum (
    val amount: Long? = null,

    @SerialName("average_price")
    val averagePrice: Long? = null,

    @SerialName("exchange_order_id")
    val exchangeOrderID: String? = null,

    @SerialName("exchange_timestamp")
    val exchangeTimestamp: String? = null,

    val folio: JsonObject? = null,
    val fund: String? = null,

    @SerialName("last_price")
    val lastPrice: Double? = null,

    @SerialName("last_price_date")
    val lastPriceDate: String? = null,

    @SerialName("order_id")
    val orderID: String? = null,

    @SerialName("order_timestamp")
    val orderTimestamp: String? = null,

    @SerialName("placed_by")
    val placedBy: String? = null,

    @SerialName("purchase_type")
    val purchaseType: String? = null,

    val quantity: Long? = null,

    @SerialName("settlement_id")
    val settlementID: String? = null,

    val status: String? = null,

    @SerialName("status_message")
    val statusMessage: String? = null,

    val tag: String? = null,
    val tradingsymbol: String? = null,

    @SerialName("transaction_type")
    val transactionType: String? = null,

    val variety: String? = null
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/mf_orders.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Debug, Serialize, Deserialize)]
pub struct MfOrders {
    pub data: Option<Vec<Datum>>,
    pub status: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Datum {
    pub amount: Option<i64>,
    pub average_price: Option<i64>,
    pub exchange_order_id: Option<String>,
    pub exchange_timestamp: Option<String>,
    pub folio: Option<serde_json::Value>,
    pub fund: Option<String>,
    pub last_price: Option<f64>,
    pub last_price_date: Option<String>,
    pub order_id: Option<String>,
    pub order_timestamp: Option<String>,
    pub placed_by: Option<String>,
    pub purchase_type: Option<String>,
    pub quantity: Option<i64>,
    pub settlement_id: Option<String>,
    pub status: Option<String>,
    pub status_message: Option<String>,
    pub tag: Option<String>,
    pub tradingsymbol: Option<String>,
    pub transaction_type: Option<String>,
    pub variety: Option<String>,
}
